src/inc/CombFilterIf.h

```diff
@@ -0,0 +1,113 @@
+#if !defined(__CombFilterIf_hdr__)
+#define __CombFilterIf_hdr__
+
+#include "ErrorDef.h"
+
+class CCombFilterBase; // in case you intend to add an internal base class that the user doesn't see (not required)
+
+/*! \brief interface class for the comb filter (FIR & IIR)
+*/
+class CCombFilterIf
```

Comment:
Most, if not all of the methods within this class have not been implemented in the cpp file, but instead return only an Error_t value rather than doing what they are supposed to